src/sentry/tasks/summaries/daily_summary.py

```diff
@@ -111,13 +119,24 @@ def prepare_summary_data(
     timestamp: float,
     duration: int,
     organization_id: int,
+    users_to_send_to: list[int],
 ):
     organization = Organization.objects.get(id=organization_id)
     ctx = build_summary_data(
         timestamp=timestamp, duration=duration, organization=organization, daily=True
     )
-    return ctx
-    # XXX(CEO): in a follow up PR this will delivery the daily summary
+    with sentry_sdk.start_span(op="daily_summary.check_if_ctx_is_empty"):
+        report_is_available = not check_if_ctx_is_empty(ctx)
+    set_tag("report.available", report_is_available)
+
+    if not report_is_available:
+        logger.info(
+            "prepare_organization_report.skipping_empty", extra={"organization": organization_id}
+        )
+        return
+
+    with sentry_sdk.start_span(op="daily_summary.deliver_summary"):
+        deliver_summary(ctx=ctx, users=users_to_send_to)
```

Comment:
@ceorourke seems odd to me that `prepare_summary_data` is now responsible for delivering the summaries as well

Reply:
Is it? I'm following the same pattern weekly reports follows - it builds up the context and then sends the message.

Reply:
if we do mutations then this is probably better so we can leave as is

Reply:
@ceorourke got it